builtin/docker/task.go

```diff
@@ -0,0 +1,369 @@
+	// See the docker docs for more info: https://docs.docker.com/config/labels-custom-metadata/
+	Labels map[string]string `hcl:"labels,optional"`
+
+	// An array of strings with network names to connect the container to
+	Networks []string `hcl:"networks,optional"`
+
+	// Resources configures the resource constraings such as cpu and memory for the
+	// created containers.
+	Resources TaskResources `hcl:"resources,block"`
+
+	// Environment variables that are meant to configure the application in a static
+	// way. This might be control an image that has mulitple modes of operation,
+	// selected via environment variable. Most configuration should use the waypoint
+	// config commands.
+	StaticEnvVars map[string]string `hcl:"static_environment,optional"`
+}
+
+func (b *TaskLauncher) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(
+		docs.FromConfig(&TaskLauncherConfig{}),
+		docs.FromFunc(b.StartTaskFunc()),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description(`
+Launch a Docker container as a task.
+
+If a Docker server is available (either locally or via environment variables
+such as "DOCKER_HOST"), then it will be used to start the container.
+`)
+
+	doc.Example(`
+task {
+  use "docker" {
+		force_pull = true
+  }
+}
+`)
+
+	doc.SetField(
+		"binds",
+		"A 'source:destination' list of folders to mount onto the container from the host.",
+		docs.Summary(
+			"A list of folders to mount onto the container from the host. The expected",
+			"format for each string entry in the list is `source:destination`. So",
+			"for example: `binds: [\"host_folder/scripts:/scripts\"]",
+		),
+	)
+
+	doc.SetField(
+		"labels",
+		"A map of key/value pairs to label the docker container with.",
+		docs.Summary(
+			"A map of key/value pair(s), stored in docker as a string. Each key/value pair must",
+			"be unique. Validiation occurs at the docker layer, not in Waypoint. Label",
+			"keys are alphanumeric strings which may contain periods (.) and hyphens (-).",
+		),
+	)
+
+	doc.SetField(
+		"networks",
+		"An list of strings with network names to connect the container to.",
+		docs.Default("waypoint"),
+		docs.Summary(
+			"A list of networks to connect the container to. By default the container",
+			"will always connect to the `waypoint` network.",
+		),
+	)
+
+	doc.SetField(
+		"resources",
+		"The resources that the tasks should use.",
+		docs.SubFields(func(d *docs.SubFieldDoc) {
+			d.SetField("cpu", "The cpu shares that the tasks should use")
+			d.SetField("memory", "The amount of memory to use. Defined as '512MB', '44kB', etc.")
+		}),
+	)
+
+	doc.SetField(
+		"static_environment",
+		"environment variables to expose to the application",
+		docs.Summary(
+			"these environment variables should not be run of the mill",
+			"configuration variables, use waypoint config for that.",
+			"These variables are used to control over all container modes,",
+			"such as configuring it to start a web app vs a background worker",
```

Comment:
```suggestion
			"These variables are used to control all of a container's modes,",
			"such as configuring it to start a web app vs a background worker.",
			"These environment variables should not be common",
			"configuration variables normally set in `waypoint config`.",
```
Rearranged this a bit. I feel like the purpose should go first and the anti-purpose should be a clarifier. Also removed the slang term that, tbh, kind of confused me 😅

Reply:
For ease of copy/pasting if you don't want to use `Commit suggestion` 😄 
```
			"These variables are used to control all of a container's modes,",
			"such as configuring it to start a web app vs a background worker.",
			"These environment variables should not be common",
			"configuration variables normally set in `waypoint config`.",
```